src/scales/scale.time.js

```diff
@@ -403,6 +408,21 @@ function ticksFromTimestamps(values, majorUnit) {
 	return ticks;
 }
 
+/**
+ * Show the most specific time format by default
+ */
+function defaultTooltipFormat(context) {
```

Comment:
I understand it's the default you want for the financial chart but I'm not sure about using it as default for all charts. I usually prefer consistency when reading data and IMO having the tooltip display format changing for all values is not a good user experience. I would go for a static default (maybe the most detailed `'MMM D, YYYY h:mm:ss.SSS a'`) - this would also avoid converting, *by default*, 2 times the same value to a Moment object (though that's minor).

Reply:
I slept on it and finally, I think the best default behavior is to keep `tooltipFormat` undefined and handle this case in `getLabelForIndex` as: if `tooltipFormat` is null or undef, use the same display format as the one chosen for the scale labels, so *by default*, the tooltip displays exactly the same text as the tick label (I remember a ticket or comment requesting that behavior).

Reply:
Part of what I was trying to accomplish here was to show more detail in the tooltip than is on the scale as that's a common use of tooltips that I think people expect. E.g. if the scale says "May 11" then it would be nice that when you mouseover with the tooltip that you see "May 11, 2017". I think using the same display format as the one chosen for the scale labels is not a bad choice either. Perhaps we could have some way to toggle between these two behaviors? E.g. define the `tooltipFormat` as this function, but then if you set it to `null` then it displays the same as the scale or something like that.

Reply:
My issue with that method is that the tooltip displays different formats depending on the current value, which I think it's not a good UX. It can be different from the tick label (more detailed) but should be the same for all the values, so when the user moves his mouse, he always reads the same format.

That's why I would implement the way I described in my second comment since it's simpler, more optimized and I think more expectable as a default.

Though that would be great to have more feedback.